src/main/java/com/currencycloud/client/CurrencyCloudClient.java

```diff
@@ -253,7 +241,12 @@ public Balance retrieveBalance(String currency) throws CurrencyCloudException {
     ///////////////////////////////////////////////////////////////////
     ///// BENEFICIARIES ///////////////////////////////////////////////
 
+    @Override
     public Beneficiary validateBeneficiary(Beneficiary beneficiary) throws CurrencyCloudException {
+        return validateBeneficiary(beneficiary, null);
+    }
+
+    protected Beneficiary validateBeneficiary(Beneficiary beneficiary, String onBehalfOf) {
```

Comment:
Some functions seem to have onbehalf within the new args but not referenced in the function. The similar old functions did not have onbehalf referenced I assume because it isn't supported by that action. The new setup might, therefore, be misleading because it might imply it is supported if someone does not look at the function implementation.

Reply:
This is one example of this new setup where it isn't referenced in the function. There are a few others too. One possible solution is to only support the new setup on functions that support onbehalf.

Reply:
@abhipatel-cc All new functions with `onBehalfOn` parameter are protected and NOT publicly available on `CurrencyCloudClient`. Users have to use `OnBehalfClient` (created with `onBehalfOf` method) to invoke any on-behalf function, e.g.:
```java
client.onBehalfOf(contactId).validateBeneficiary(beneficiary);
```
or
```java
client.onBehalfOfDo(contactId, new OnBehalfRunnable() {
    @Override
    public void run(OnBehalfClient onBehalfClient) {
        onBehalfClient.validateBeneficiary(beneficiary);
    }
});
```
Running `validateBeneficiary` on the client object will always execute with onBehalf == null.